homeassistant/components/sense/entity.py

```diff
@@ -0,0 +1,72 @@
+"""Base entities for Sense energy."""
+
+from sense_energy import ASyncSenseable
+from sense_energy.sense_api import SenseDevice
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTRIBUTION, DOMAIN, MDI_ICONS
+from .coordinator import SenseCoordinator
+
+
+def sense_to_mdi(sense_icon: str) -> str:
+    """Convert sense icon to mdi icon."""
+    return f"mdi:{MDI_ICONS.get(sense_icon, "power-plug")}"
+
+
+class SenseEntity(CoordinatorEntity[SenseCoordinator]):
+    """Base implementation of a Sense sensor."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: SenseCoordinator,
+        sense_monitor_id: str,
+        unique_id: str,
+        gateway: ASyncSenseable,
```

Comment:
Personal preference, have the big parameters at the front and the smaller parameters like `str` at the back